src/main/java/dan200/computercraft/shared/util/Palette.java

```diff
@@ -120,4 +122,19 @@ public void readFromNBT( NBTTagCompound nbt )
             colours[i] = decodeRGB8( rgb8[i] );
         }
     }
+
+    @Override
+    public boolean equals( Object o )
+    {
+        if( this == o ) return true;
+        if( o == null || getClass() != o.getClass() ) return false;
+        Palette palette = (Palette) o;
+        return Arrays.equals( colours, palette.colours );
```

Comment:
This line will always return `false`. It checks reference equality for each colour's `double[]`, but those are unique for each palette.

Reply:
`equals` and `hashCode` aren't supposed to be here anyway, they're a remnant of when I was working on something else.